internal/resource/resource_service.go

```diff
@@ -76,21 +129,90 @@ func (r *ResourceService) DeleteInstances(instanceList []*v1.Instance) *v1.Resou
 			}
 		}
 	}
+	r.RemoveOperator(instanceList)
 
 	return r.resource
 }
 
+func (r *ResourceService) Apply(ctx context.Context, instanceID string) error {
+	var instance *mpi.Instance
+	var errorsFound error
+	operator := r.instanceOperators[instanceID]
+
+	for _, resourceInstance := range r.resource.GetInstances() {
+		if resourceInstance.GetInstanceMeta().GetInstanceId() == instanceID {
+			instance = resourceInstance
+		}
+	}
+
+	valErr := operator.Validate(ctx, instance)
+	if valErr != nil {
+		return fmt.Errorf("failed validating config %w", valErr)
+	}
+
+	reloadErr := operator.Reload(ctx, instance)
+	if reloadErr != nil {
+		return fmt.Errorf("failed to reload NGINX %w", reloadErr)
+	}
+
+	errorLogs := r.errorLogs(instance)
```

Comment:
I think the error log checks should move to the operator Reload function since you need to start tailing the logs just before you perform the reload. Plus this error log checking is specific to only NGINX and not other instances like unit for example.

Reply:
like all of the error log logic ?

Reply:
yeah I think it all needs to go into the reload function